pandas/core/arrays/datetimes.py

```diff
@@ -307,7 +307,12 @@ def _generate_range(cls, start, end, periods, freq, tz=None,
                     end = end.tz_localize(tz).asm8
         else:
             # Create a linearly spaced date_range in local time
-            arr = np.linspace(start.value, end.value, periods)
+            # Nanosecond-granularity timestamps aren't always correctly
+            # representable with doubles, so we limit the range that we
+            # pass to np.linspace as much as possible
+            arr = np.linspace(
+                0, end.value - start.value,
+                periods).astype('int64') + start.value
```

Comment:
You can just pass in `int64` into the dtype argument of `linspace` instead of `astype`ing here.

Reply:
I’ve been wrestling with something similar (#24123).  Does this get fixed if you cast “endpoint” to int64 in generate_range_overflow_safe?

Reply:
@jbrockmendel generate_range_overflow_save isn't getting called in this situation, because no frequency has been supplied (which I assume is why np.linspace is used).